### PR TITLE
gio: add AsyncInitable

### PR DESCRIPTION
**Pull request by pbor**

### CODE REVIEW

File: gio/src/async_initable.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+use crate::Cancellable;
+use glib::object::IsA;
+use glib::object::IsClass;
+use glib::value::ToValue;
+use glib::{Cast, Object, StaticType, Type};
+
+impl AsyncInitable {
+    #[doc(alias = "g_async_initable_new_async")]
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new_async<
```
Comment (bilelmoussaoui) [inline gio/src/async_initable.rs:14]: Should probably have a _future variant as well?

File: gio/Gir.toml [hunk 2]
```diff
@@ -307,6 +307,19 @@ status = "generate"
         name = "name"
         string_type = "os_string"
 
+[[object]]
+name = "Gio.AsyncInitable"
```
Comment (bilelmoussaoui) [inline gio/Gir.toml:311]: Should also be added to the other crates in the `manual` list so that it is implemented for the types that uses it

Comment (pbor) [reply gio/Gir.toml:311]: Which ones do you have in mind? as far as I know gtk etc does not have any async initable, no?

File: gio/src/async_initable.rs (hunk 1 above)
Comment (pbor) [reply gio/src/async_initable.rs:14]: It should... but that always makes my head spin :)

File: gio/Gir.toml (hunk 2 above)
Comment (bilelmoussaoui) [reply gio/Gir.toml:311]: Never mind, i didn't found any in gtk-rs-core/gtk3-rs/gtk4-rs

File: gio/src/auto/async_initable.rs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    }
+}
+
+impl AsyncInitable {
+    pub const NONE: Option<&'static AsyncInitable> = None;
+}
+
+pub trait AsyncInitableExt: 'static {
+    #[doc(alias = "g_async_initable_init_async")]
+    fn init_async<P: FnOnce(Result<(), glib::Error>) + 'static>(
```
Comment (sdroege) [inline gio/src/auto/async_initable.rs:29]: This needs to be `unsafe` as you are not allowed to call it multiple times on an object. Also the future variant below

**Review by sdroege — COMMENTED** [review]
Otherwise seems good to me

File: gio/src/auto/async_initable.rs (hunk 3 above)
Comment (pbor) [reply gio/src/auto/async_initable.rs:29]: If I mark `init_async` as unsafe in Gir.toml, not only `init_future` is not automatically marked as unsafe, but it fails to compile. This seems to be a bug in gir.

**Review by sdroege — APPROVED** [review]

**Pull request merged**